solr/core/src/java/org/apache/solr/handler/component/RealTimeGetComponent.java

```diff
@@ -296,13 +297,13 @@ public void process(ResponseBuilder rb) throws IOException
          if (null == resultContext) {
            // either first pass, or we've re-opened searcher - either way now we setContext
            resultContext = new RTGResultContext(rsp.getReturnFields(), searcherInfo.getSearcher(), req);
-           transformer.setContext(resultContext);
+           transformer.prepare(resultContext);
          }
          transformer.transform(doc, docid, 0);
        }
        docList.add(doc);
      }
-
+     if ( null != transformer) transformer.finish();
```

Comment:
if it impacts RTG it also should be tested, otherwise, it can lead to leak from some heavy transformer